File: native-ios/JuanbertosPOS/App/AppState.swift
import SwiftUI

@Observable
final class AppState {
    var currentEmployee: Employee?
    var currentScreen: Screen = .login

    var isLoggedIn: Bool { currentEmployee != nil }

    func navigate(to screen: Screen) {
        withAnimation(.easeInOut(duration: 0.2)) {
            currentScreen = screen
        }
    }

    func logout() {
        currentEmployee = nil
        currentScreen = .login
    }

    func loginSucceeded(employee: Employee) {
        currentEmployee = employee
        // Kitchen staff go straight to kitchen display
        if employee.role == .kitchen {
            currentScreen = .kitchen
        } else {
            currentScreen = .pos
        }
    }
}

enum Screen: Sendable {
    case login
    case pos
    case kitchen
    case admin
    case reports
    case inventory
    case employees
    case menuManagement
}

File: apps/pos/native-android/app/src/main/java/com/desktopkitchen/pos/app/AppState.kt
package com.desktopkitchen.pos.app

import androidx.compose.runtime.getValue
import androidx.compose.runtime.mutableStateOf
import androidx.compose.runtime.setValue
import com.desktopkitchen.pos.models.Employee
import com.desktopkitchen.pos.networking.AuthTokenStore
import javax.inject.Inject
import javax.inject.Singleton

enum class Screen {
    Login,
    POS,
    Kitchen,
    Admin,
    Reports,
    Inventory,
    Employees,
    MenuManagement,
    Kiosk,
    OrderHistory
}

@Singleton
class AppState @Inject constructor() {
    var currentScreen by mutableStateOf(Screen.Login)
        private set
    var currentEmployee by mutableStateOf<Employee?>(null)
        private set

    val isLoggedIn: Boolean get() = currentEmployee != null

    fun navigate(screen: Screen) {
        currentScreen = screen
    }

    fun loginSucceeded(employee: Employee) {
        currentEmployee = employee
        AuthTokenStore.token = employee.token
        currentScreen = if (employee.role == "kitchen") Screen.Kitchen else Screen.POS
    }

    fun logout() {
        currentEmployee = null
        AuthTokenStore.token = null
        currentScreen = Screen.Login
    }
}
